Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Person.hs

```diff
@@ -264,14 +273,15 @@ updateBlockedState personId isBlocked = do
       ]
     where_ $ tbl ^. PersonId ==. val (getId personId)
 
-updatingEnabledAndBlockedState :: Id Person -> Bool -> SqlDB ()
-updatingEnabledAndBlockedState personId isBlocked = do
+updatingEnabledAndBlockedState :: Id Person -> Maybe (Id DMC.MerchantConfig) -> Bool -> SqlDB ()
+updatingEnabledAndBlockedState personId blockedByRule isBlocked = do
```

Comment:
`blockedByRule` -  can this be a combination? mb `searchCount >= searchThreshold` and `totalRides <= ridesThreshold`? If not, I don't see `TotalRides` as a single factor for any of the blockings

Reply:
`blockedByRule` is the id from merchant_config table, this will help us understand which of the multiple rules was used to block a particular customer. (as we are having multiple rules now)